### DISCUSSION TITLE
Unclear behavior of Unloaded event for UserControl

### DISCUSSION THREAD

**Discussion by chornyi-xrs** [AvaloniaUI/Avalonia#15464] [Q&A]
Good afternoon everyone, I assume there is a problem with the occurrence of Uloaded event. I tried to use the following code:
```
<Interaction.Behaviors>
  <EventTriggerBehavior EventName="Loaded">
	  <InvokeCommandAction Command="{Binding LoadedCommand}" />
  </EventTriggerBehavior>
  <EventTriggerBehavior EventName="Unloaded">
	  <InvokeCommandAction Command="{Binding UnloadedCommand}" />
  </EventTriggerBehavior>
</Interaction.Behaviors>
```
Only the Loaded command works.

But when you change the code to the following:
```
<Interaction.Behaviors>
  <EventTriggerBehavior EventName="AttachedToVisualTree">
	  <InvokeCommandAction Command="{Binding LoadedCommand}" />
  </EventTriggerBehavior>
  <EventTriggerBehavior EventName="DetachedFromVisualTree">
	  <InvokeCommandAction Command="{Binding UnloadedCommand}" />
  </EventTriggerBehavior>
</Interaction.Behaviors>
```
Everything is starting to work as I expect it to.

Is this a known problem or am I missing something?

**Comment by timunie** [comment]
without seeing a minimal sample, no chance to help here.

**Comment by robloo** [comment]
`Unloaded` is invoked directly from the `DetachedFromVisualTree` code in control. They should behave exactly the same. The only differences are in `Loaded`

**Comment by maxkatz6** [comment]
Should be the same, but either way you _should_ use RoutedEventTriggerBehavior for routed events like Loaded/Unloaded as per Behaviors SDK.
